pyensign/ensign.py

```diff
@@ -118,29 +132,62 @@ async def next():
             on_nack=on_nack,
         )
 
-    async def subscribe(self, *topic_ids, query="", consumer_group=None):
+    async def subscribe(self, *topics, query="", consumer_group=None):
         """
         Subscribe to events from the Ensign server.
 
         Parameters
         ----------
         topic_ids : iterable of str
-            The topic IDs to subscribe to.
+            The topic names or IDs to subscribe to.
 
         query : str (optional)
             EnSQL query to filter events.
 
         consumer_group : api.v1beta1.groups.ConsumerGroup (optional)
             The consumer group to use for the subscriber.
 
+        Raises
+        ------
+        ValueError
+            If no topics are provided.
+
+        EnsignTopicNotFoundError
+            If a topic name is provided that does not exist.
+
         Yields
         ------
         api.v1beta1.event_pb2.Event
             The events received from the Ensign server.
         """
 
-        if len(topic_ids) == 0:
-            raise ValueError("no topic IDs provided")
+        # Handle a list of topics passed as a single argument
+        if len(topics) == 1 and (
+            isinstance(topics[0], list) or isinstance(topics[0], tuple)
+        ):
+            topics = topics[0]
+
+        if len(topics) == 0:
+            raise ValueError("no topics provided")
+
+        # Ensure all topics are strings
+        for topic in topics:
+            if not isinstance(topic, str):
+                raise TypeError(
+                    "expected type 'str' for topic, got {}".format(type(topic))
+                )
+
+        # Parse topic names into ID strings
+        topic_ids = []
+        for topic in topics:
+            # First try to parse as a ULID
+            try:
+                id = ULID().from_str(topic)
```

Comment:
This doesn't check if the topic exists if a topic ID is provided, although we probably should. I'm working on an update right now.